### ISSUE TITLE
Black screen on win shadow server.

### ISSUE THREAD

**Issue by jkustosz** [FreeRDP/FreeRDP#3060]
Hi,

Based on the master branch I've compiled and built the freerdp-shadow-cli.exe on Windows 7 with Visual.

When trying to connect to Win shadow server from linux client with 
xfreerdp /v:server_ip:port /cert-ignore
I have black screen. I tried several option with RemoteFX and ColorDepth but this did not help.
When connected from the iFreeRDP iphone application also get the black screen but I was able to move the mouse coursor and use the keyboard. 

When using the mstsc connection - the same. 
Any help? Thanks,

jarek

**Comment by jkustosz** [comment]
I just realized that the problem is related to the BIO_read function.

During debugging I found that when shadow server is starting ( on port e.g. 5555)  it trying to connect to the the another port 40000. But this negotiation fails because of the BIO_read function failed.
function  transport_read_layer( ) calls  BIO_read and BIO_should retry and those functions fails. It produce the error in the log:

[14:31:30:843] [10632:00003afc] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
14:31:30:844] [10632:00003afc] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x2000C]
[14:31:30:844] [10632:00003afc] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
[14:31:30:845] [10632:00003afc] [INFO][com.freerdp.server.shadow.win] - OnConnectionResult: -1

This happened before any client is connected - just during execution of shadow server on windows.

Does anyone knows why BIO_read and BIO_should_retry() can not read any data?

Is this related to openssl ?

**Comment by realjiangms** [comment]
I always get black screen too. The shadow server for Win  is very experimental. It seems need DirectX 11 but I don't have such environment. So it just work at protocol level (Connection is OK and operations from client side could be sent to server correctly).

However I always get following error which is different:
[12:31:23:295] [2151744:0020d5a0] [ERROR][com.freerdp.server.shadow.win] - IRDPS
RAPISharingSession::put_ColorDepth() failure: 0x80070057
[12:31:23:297] [2151744:0020d5a0] [INFO][com.freerdp.server.shadow.win] - width:
 0 height: 0
[12:31:23:314] [2151744:0020d5a0] [INFO][com.freerdp.core.listener] - Listening
on :::4489

Not sure what port 40000 means. Does your environment enabled DX11?

**Comment by 0nur0zcan** [comment]
Today, I have downloaded FreeRDP-2.0.0-rc0 on my Win OS. Then, I've run freerdp-shadow-cli-exe and I got the same errors, as @realjiangms indicated above. And also when I've started wfreerdp.exe I've got black screen, too. How can I make them run correctly?

**Comment by jc0456** [comment]
Hi,
I use the master branch and run freerdp-shadow-cli-exe on Windows 7 Ultimate with Service Pack 1 (x64)(Chinese-Simplified),then I got the same errors as @jkustosz indicated above.
I find the RDPSession was ok , The TCP link between shw_client and rdpsession has been established.but the Rdpsession has broken the TCP link after receiving the x.224 message sent by shw_client. I checked the x.224 message sent by shw_client and compared it with the Protocol Examples and found that it was correct.
I run the same freerdp-shadow-cli-exe on Windows 10 and it was ok.I use the same compiled wfreerdp-exe to connect with the remote desktop system owned by my Windows 7 computer itself and the nego_recv_response  function was correct.
I don't know why. I wonder whether there is something wrong with my computer. Can you give me some good advice?
thanks!

**Comment by aijyo** [comment]
> 4 message sent by shw_client. I checked the x.224 message sent by shw_client and compared it with the Protocol Examples and found that it was correct.
> I run the same freerdp-shadow-cli-exe on Windows 10 and it was ok.I us

你也在windows上运行freerdp 服务端吗？  弄好了吗，我最近也遇到好多问题 想问下你...

**Comment by jc0456** [comment]
> > 4 message sent by shw_client. I checked the x.224 message sent by shw_client and compared it with the Protocol Examples and found that it was correct.
> > I run the same freerdp-shadow-cli-exe on Windows 10 and it was ok.I us
> 
> 你也在windows上运行freerdp 服务端吗？ 弄好了吗，我最近也遇到好多问题 想问下你...

No,wfreerdp-server instead, https://github.com/akallabeth/FreeRDP/tree/dsp_audio_server_api
install dfmirage, http://www.demoforge.com/dfmirage.htm

**Comment by aijyo** [comment]
> > > 4 message sent by shw_client. I checked the x.224 message sent by shw_client and compared it with the Protocol Examples and found that it was correct.
> > > I run the same freerdp-shadow-cli-exe on Windows 10 and it was ok.I us
> > 
> > 
> > 你也在windows上运行freerdp 服务端吗？ 弄好了吗，我最近也遇到好多问题 想问下你...
> 
> No,wfreerdp-server instead, https://github.com/akallabeth/FreeRDP/tree/dsp_audio_server_api
> install dfmirage, http://www.demoforge.com/dfmirage.htm

ah? You don't use wfreerdp-server, but instead use dfmirage instead?

**Comment by jc0456** [comment]
> > > > 4 message sent by shw_client. I checked the x.224 message sent by shw_client and compared it with the Protocol Examples and found that it was correct.
> > > > I run the same freerdp-shadow-cli-exe on Windows 10 and it was ok.I us
> > > 
> > > 
> > > 你也在windows上运行freerdp 服务端吗？ 弄好了吗，我最近也遇到好多问题 想问下你...
> > 
> > 
> > No,wfreerdp-server instead, https://github.com/akallabeth/FreeRDP/tree/dsp_audio_server_api
> > install dfmirage, http://www.demoforge.com/dfmirage.htm
> 
> ah? You don't use wfreerdp-server, but instead use dfmirage instead?

shadow的还没有找到问题在哪里，后来换成了wfreerdp-server，后面那个地址是原作者解决了wfreerdp-server不能编译的问题的可编译包，使用wfreerdp-server必须要安装那个dfmirage 显卡镜像驱动，后面附了下载地址

**Comment by aijyo** [comment]
> > > > > 4 message sent by shw_client. I checked the x.224 message sent by shw_client and compared it with the Protocol Examples and found that it was correct.
> > > > > I run the same freerdp-shadow-cli-exe on Windows 10 and it was ok.I us
> > > > 
> > > > 
> > > > 你也在windows上运行freerdp 服务端吗？ 弄好了吗，我最近也遇到好多问题 想问下你...
> > > 
> > > 
> > > No,wfreerdp-server instead, https://github.com/akallabeth/FreeRDP/tree/dsp_audio_server_api
> > > install dfmirage, http://www.demoforge.com/dfmirage.htm
> > 
> > 
> > ah? You don't use wfreerdp-server, but instead use dfmirage instead?
> 
> shadow的还没有找到问题在哪里，后来换成了wfreerdp-server，后面那个地址是原作者解决了wfreerdp-server不能编译的问题的可编译包，使用wfreerdp-server必须要安装那个dfmirage 显卡镜像驱动，后面附了下载地址

感谢，我跟作者聊了，shadow还没做好的，所以有毛病，我试试你说的wfreerdp-server
